nbconvert/nbconvertapp.py

```diff
@@ -30,6 +31,36 @@
 #Classes and functions
 #-----------------------------------------------------------------------------
 
+class CaselessStrEnumOrDottedObjectName(CaselessStrEnum, DottedObjectName):
```

Comment:
You can provide a nicer error message in the code below where we're setting up `self.exporter`, rather than hammering traitlets into giving a roughly right but rather ugly error.